src/nvim/edit.c

```diff
@@ -3530,21 +3530,20 @@ int ins_compl_add_tv(typval_T *tv, int dir)
   char_u      *(cptext[CPT_COUNT]);
 
   if (tv->v_type == VAR_DICT && tv->vval.v_dict != NULL) {
-    word = get_dict_string(tv->vval.v_dict, (char_u *)"word", FALSE);
-    cptext[CPT_ABBR] = get_dict_string(tv->vval.v_dict,
-        (char_u *)"abbr", FALSE);
-    cptext[CPT_MENU] = get_dict_string(tv->vval.v_dict,
-        (char_u *)"menu", FALSE);
-    cptext[CPT_KIND] = get_dict_string(tv->vval.v_dict,
-        (char_u *)"kind", FALSE);
-    cptext[CPT_INFO] = get_dict_string(tv->vval.v_dict,
-        (char_u *)"info", FALSE);
-    if (get_dict_string(tv->vval.v_dict, (char_u *)"icase", FALSE) != NULL)
-      icase = get_dict_number(tv->vval.v_dict, (char_u *)"icase");
-    if (get_dict_string(tv->vval.v_dict, (char_u *)"dup", FALSE) != NULL)
-      adup = get_dict_number(tv->vval.v_dict, (char_u *)"dup");
-    if (get_dict_string(tv->vval.v_dict, (char_u *)"empty", FALSE) != NULL)
-      aempty = get_dict_number(tv->vval.v_dict, (char_u *)"empty");
+    word = get_dict_string(tv->vval.v_dict, "word", false);
+    cptext[CPT_ABBR] = get_dict_string(tv->vval.v_dict, "abbr", false);
+    cptext[CPT_MENU] = get_dict_string(tv->vval.v_dict, "menu", false);
+    cptext[CPT_KIND] = get_dict_string(tv->vval.v_dict, "kind", false);
+    cptext[CPT_INFO] = get_dict_string(tv->vval.v_dict, "info", false);
+    if (get_dict_string(tv->vval.v_dict, "icase", false) != NULL) {
```

Comment:
Though only `icase = get_dict_number(tv->vval.v_dict, "icase")` should be enough. Booleans defaulting to false/zero is the general pattern, if needed a comment could be added.